samples/dnn/yolo_object_detection.cpp

```diff
@@ -0,0 +1,123 @@
+#include <opencv2/dnn.hpp>
+#include <opencv2/dnn/shape_utils.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/highgui.hpp>
+using namespace cv;
+using namespace cv::dnn;
+
+#include <fstream>
+#include <iostream>
+#include <algorithm>
+#include <cstdlib>
+using namespace std;
+
+const size_t network_width = 416;
+const size_t network_height = 416;
+
+static cv::Mat preprocess(const cv::Mat& frame, int width, int height)
+{
+    cv::Mat preprocessed;
+    cv::resize(frame, preprocessed, cv::Size(width, height)); //Yolo accepts 416x416 RGB-images, resolution that set in cfg
+    preprocessed.convertTo(preprocessed, CV_32F, 1 / 255.F);
+    return preprocessed;
+}
+
+const char* about = "This sample uses You only look once (YOLO)-Detector "
+                    "(https://arxiv.org/abs/1612.08242)"
+                    "to detect objects on image\n"; // TODO: link
+
+const char* params
+    = "{ help           | false | print usage         }"
+      "{ cfg            |       | model configuration }"
+      "{ model          |       | model weights       }"
+      "{ image          |       | image for detection }"
+      "{ min_confidence | 0.24  | min confidence      }";
+
+int main(int argc, char** argv)
+{
+    cv::CommandLineParser parser(argc, argv, params);
+
+    if (parser.get<bool>("help"))
+    {
+        std::cout << about << std::endl;
+        parser.printMessage();
+        return 0;
+    }
+
+    String modelConfiguration = parser.get<string>("cfg");
+    String modelBinary = parser.get<string>("model");
+
+    //! [Initialize network]
+    dnn::Net net = readNetFromDarknet(modelConfiguration, modelBinary);
+    //! [Initialize network]
+
+    if (net.empty())
+    {
+        cerr << "Can't load network by using the following files: " << endl;
+        cerr << "cfg-file:     " << modelConfiguration << endl;
+        cerr << "weights-file: " << modelBinary << endl;
+        cerr << "Models can be downloaded here:" << endl;
+        cerr << "https://pjreddie.com/darknet/yolo/" << endl;
+        exit(-1);
+    }
+
+    cv::Mat frame = cv::imread(parser.get<string>("image"), -1);
+
+    if (frame.channels() == 4)
+        cvtColor(frame, frame, cv::COLOR_BGRA2BGR);
+    //! [Prepare blob]
+    Mat preprocessedFrame = preprocess(frame, network_width, network_height);
```

Comment:
Please, use `blobFromImage`'s arguments to make preprocessing (http://docs.opencv.org/3.3.0/d6/d0f/group__dnn.html#ga0507466a789702eda8ffcdfa37f4d194).